esp-hal-common/src/serial.rs

```diff
@@ -26,7 +158,35 @@ impl<T> Serial<T>
 where
     T: Instance,
 {
-    /// Create a new UART instance
+    /// Create a new UART instance with defaults
+    pub fn new_with_config<
+        TX: OutputPin<OutputSignal = OutputSignal>,
+        RX: InputPin<InputSignal = InputSignal>,
+        CTS: InputPin<InputSignal = InputSignal>,
+        RTS: OutputPin<OutputSignal = OutputSignal>,
+    >(
+        uart: T,
+        config: Option<Config>,
+        pins: Option<Pins<TX, RX, CTS, RTS>>,
+        clocks: &Clocks,
+    ) -> Result<Self, Error> {
```

Comment:
Can this actually fail? Is the `Result<_>` return type necessary?

Reply:
No - just copied the signature from the function that used to be here 👍